### PR TITLE
Core/Misc: implemented petition manager

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Tests performed:** Compiles, ingame tests

Implemented manager for petitions to perform all petition related tasks and synchronize data with database. Review, tests and comments are welcomed.

### CODE REVIEW

File: src/server/game/Petitions/PetitionMgr.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+{
+	public:
+        PetitionMgr() { }
+        ~PetitionMgr() { }
+
+        static PetitionMgr* instance();
+
+        // Load from DB
+		void LoadPetitions();
+		void LoadSignatures();
```
Comment (Keader) [inline src/server/game/Petitions/PetitionMgr.h]: identation

File: src/server/game/Petitions/PetitionMgr.cpp [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    p.petitionGuid = petitionGuid;
+    p.ownerGuid = ownerGuid;
+    p.petitionName = name;
+    p.petitionType = type;
+    p.signatures.clear();
+
+    if (isLoading)
+        return;
+
+    SQLTransaction trans = CharacterDatabase.BeginTransaction();
```
Comment (Shauren) [inline src/server/game/Petitions/PetitionMgr.cpp]: transaction for a single query? doesn't add up

Comment (xinef1) [inline src/server/game/Petitions/PetitionMgr.cpp]: True, overlooked

File: src/server/game/Petitions/PetitionMgr.cpp [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    {
+        TC_LOG_INFO("server.loading", ">> Loaded 0 petitions.");
+        return;
+    }
+
+    uint32 count = 0;
+    do
+    {
+        Field* fields = result->Fetch();
+        AddPetition(ObjectGuid(HighGuid::Item, fields[0].GetUInt32()), ObjectGuid(HighGuid::Player, fields[1].GetUInt32()), fields[2].GetString(), static_cast<CharterTypes>(fields[3].GetUInt8()), true);
```
Comment (Shauren) [inline src/server/game/Petitions/PetitionMgr.cpp]: I removed all ObjectGuid constructors taking type as parameters from master branch, use static ObjectGuid::Create methods instead

File: src/server/game/Petitions/PetitionMgr.cpp [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        TC_LOG_INFO("server.loading", ">> Loaded 0 Petition signs!");
+        return;
+    }
+
+    uint32 count = 0;
+    do
+    {
+        Field* fields = result->Fetch();
+
+        Petition* petition = GetPetition(ObjectGuid(HighGuid::Item, fields[0].GetUInt32()));
```
Comment (Shauren) [inline src/server/game/Petitions/PetitionMgr.cpp]: same here with guid

File: src/server/game/Petitions/PetitionMgr.cpp [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        if (!petition)
+        {
+            continue;
+        }
+
+        petition->AddSignature(petition->petitionGuid, fields[1].GetUInt32(), ObjectGuid(HighGuid::Player, fields[2].GetUInt32()), true);
```
Comment (Shauren) [inline src/server/game/Petitions/PetitionMgr.cpp]: and here

**Pull request merged**